src/main/java/org/opentripplanner/routing/edgetype/StreetTransitEntityLink.java

```diff
@@ -100,24 +104,24 @@ public State traverse(State s0) {
         /* Only enter stations in CAR mode if parking is not required (kiss and ride) */
         /* Note that in arriveBy searches this is double-traversing link edges to fork the state into both WALK and CAR mode. This is an insane hack. */
         if (s0.getNonTransitMode() == TraverseMode.CAR) {
-            if (req.carPickup && !s0.isCarParked()) {
-                s1.setCarParked(true);
+            if (req.carPickup && s0.getCarPickupState() == CarPickupState.IN_CAR) {
+                s1.setTaxiState(s0.getOptions().arriveBy ? CarPickupState.WALK_TO_PICKUP : CarPickupState.WALK_FROM_DROP_OFF);
```

Comment:
The implementation between `TransitEntranceLink` and `StreetTransitLink` differs, with neither taking into account `arriveBy` searches.

We'll create another PR containing `carPickup` fixes to verify the correct behavior.